src/controllers/EvidenceUploadController.ts

```diff
@@ -34,32 +39,98 @@ export class EvidenceUploadController extends BaseController<OtherReason> {
+
         return {
             'upload-file': {
                 async handle(request: Request, response: Response): Promise<void> {
-                    await parseFormData(request, response);
-
-                    if (!request.file) {
-                        response.redirect(request.route.path);
-                        return;
-                    }
 
-                    const id = await that.fileTransferService.upload(request.file.buffer, request.file.originalname);
+                    const maxNumberOfFiles: number = Number(getEnvOrThrow('MAX_NUMBER_OF_FILES'));
 
                     const appeal: Appeal = request.session
                         .chain(_ => _.getExtraData())
                         .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
                         .map(data => data.appeal)
                         .unsafeCoerce();
 
+                    try {
+                        await parseFormData(request, response)
+                    } catch (error) {
+                        if (error.message === 'File not supported') {
+                            return await that
+                                .renderUploadError(appeal, fileNotSupportedError);
+                        } else if (error.message === 'File too large') {
+                            return await that
+                                .renderUploadError(appeal, fileTooLargeError);
+                        }
+                    }
+
+                    if (!request.file) {
+                        response.redirect(MOVED_TEMPORARILY, request.route.path);
+                        return;
+                    } else if (appeal.reasons.other.attachments &&
+                        appeal.reasons.other.attachments!.length >= maxNumberOfFiles) {
+                        return await that
+                            .renderUploadError(appeal, tooManyFilesError);
+                    }
+
+                    let id: string;
+
+                    try {
+                        id = await that.fileTransferService.upload(request.file.buffer, request.file.originalname);
+                    } catch (err) {
+                        if (err.message === 'Request failed with status code 415') {
+                            return await that
+                                .renderUploadError(appeal, fileNotSupportedError);
+                        } else {
+                            throw new Error(err.message)
+                        }
+                    }
+
                     appeal.reasons.other.attachments = [...appeal.reasons.other.attachments || [], {
                         id, name: request.file.originalname, contentType: request.file.mimetype, size: request.file.size
                     }];
 
                     await that.persistSession();
 
+                    response.redirect(request.route.path);
+                }
+            },
+            'upload-file-continue': {
```

Comment:
I did not expect this. There is built-in capability of BaseController that validates continue buttons, I suggest you use it.

It should be just matter of passing validator instance to base controller.

Reply:
ok, will give this a go

Reply:
I've created a validator instance for the button, was very simple to do :-)